uds/database/data_record/raw_data_record.py

```diff
@@ -0,0 +1,89 @@
+from typing import Optional, Tuple
+
+from .abstract_data_record import AbstractDataRecord, DataRecordType, DataRecordPhysicalValueAlias, DecodedDataRecord
+
+
+class RawDataRecord(AbstractDataRecord):
+    """Common implementation and interface for Raw Data Record."""
+
+    def __init__(self, name: str, length: int) -> None:
+        """
+        Initialization of Raw Data Record.
+
+        :param name: Name to assign to this Data Record.
+        :param length: Number of bits that this Raw Data Record is stored over.
+
+        :raise TypeError: Provided name is not str type.
+        :raise ValueError: Provided length is not a positive integer.
+        """
+        super().__init__(name)
+        if not isinstance(length, int) or length <= 0:
```

Comment:
I would do: `self.length = length`  here. Instead provide setter for `length` property. There you can handle type and value errors. Then you would handle all improper types and values assignment, not just on object creation.

You can base on https://github.com/mdabrowski1990/uds/blob/main/uds/transport_interface/can/common.py (e.g. `n_as_timeout` property)